game/sourcemod/scripting/donationcontrol.sp

```diff
@@ -4,187 +4,183 @@
-	hConVars[4] = CreateConVar("sm_kpanel_advertisement_time", "120.0", "Timer between messages: (1.0 + )", FCVAR_PLUGIN, true, 1.0);
-	hConVars[5] = CreateConVar("sm_kpanel_minimum", "4", "Minimum amount to donate: (Default: 4, Less than or equal to)", FCVAR_PLUGIN, true, 1.0);
+	hConVars[2] = CreateConVar("sm_kpanel_fullscreen", "1", "Enable or disable fullscreen windows", _, true, 0.0, true, 1.0);
+	hConVars[3] = CreateConVar("sm_kpanel_advertisement", "1", "Display plugin creator advertisement: (1 = on, 0 = off)", _, true, 0.0, true, 1.0);
+	hConVars[4] = CreateConVar("sm_kpanel_advertisement_time", "120.0", "Timer between messages: (1.0 + )", _, true, 1.0);
+	hConVars[5] = CreateConVar("sm_kpanel_minimum", "4", "Minimum amount to donate: (Default: 4, Less than or equal to)", _, true, 1.0);
 
-	for (new i = 0; i < sizeof(hConVars); i++)
+	for (int i = 0; i < sizeof(hConVars); i++)
 	{
-		HookConVarChange(hConVars[i], HandleCvars);
+		hConVars[i].AddChangeHook(HandleCvars);
 	}
 
 	RegConsoleCmd("sm_donate", DonatePanel);
 
-	//AutoExecConfig(true, "plugin.donation.control");
+	AutoExecConfig();
 }
 
-public OnConfigsExecuted()
+public void OnConfigsExecuted()
 {
-	cv_bEnabled = GetConVarBool(hConVars[0]);
-	GetConVarString(hConVars[1], cv_sURL, sizeof(cv_sURL));
-	cv_bFullScreen = GetConVarBool(hConVars[2]);
-	cv_bAdvertStatus = GetConVarBool(hConVars[3]);
-	cv_fAdvertTime = GetConVarFloat(hConVars[4]);
-	cv_iMinimumAmount = GetConVarInt(hConVars[5]);
+	cv_bEnabled 		= hConVars[0].BoolValue;
+	hConVars[1].GetString(cv_sURL, sizeof(cv_sURL));
+	cv_bFullScreen 		= hConVars[2].BoolValue;
+	cv_bAdvertStatus 	= hConVars[3].BoolValue;
+	cv_fAdvertTime 		= hConVars[4].FloatValue;
+	cv_iMinimumAmount 	= hConVars[5].IntValue;
 
 	if (cv_bEnabled)
 	{
 		if (cv_bAdvertStatus && cv_fAdvertTime > 1.0)
 		{
 			ClearTimer(hAdvertTimer);
-			hAdvertTimer = CreateTimer(cv_fAdvertTime, TimerAdvertisement, INVALID_HANDLE, TIMER_REPEAT);
+			hAdvertTimer = CreateTimer(cv_fAdvertTime, TimerAdvertisement, _, TIMER_REPEAT);
 		}
 
-		hDisplayMenu = CreateMenu(MenuHandle);
-
-		SetMenuTitle(hDisplayMenu, "%s", "Menu Title");
-
-		AddMenuItem(hDisplayMenu, "5", "5");
-		AddMenuItem(hDisplayMenu, "10", "10");
-		AddMenuItem(hDisplayMenu, "15", "15");
-		AddMenuItem(hDisplayMenu, "20", "20");
-
-		SetMenuExitButton(hDisplayMenu, true);
+		hDisplayMenu = new Menu(MenuHandle);
+		hDisplayMenu.SetTitle("%s", "Menu Title");
+		hDisplayMenu.AddItem("5", "5");
+		hDisplayMenu.AddItem("10", "10");
+		hDisplayMenu.AddItem("15", "15");
+		hDisplayMenu.AddItem("20", "20");
 	}
 }
 
-public HandleCvars (Handle:cvar, const String:sOldValue[], const String:sNewValue[])
+public void HandleCvars(ConVar cvar, const char[] sOldValue, const char[] sNewValue)
 {
-	if (StrEqual(sOldValue, sNewValue, true)) return;
+	if (StrEqual(sOldValue, sNewValue, true))
+		return;
 
-	new iNewValue = StringToInt(sNewValue);
+	int iNewValue = StringToInt(sNewValue);
 
 	if (cvar == hConVars[0])
```

Comment:
While here, bonus points if you convert these to switch cases

Reply:
You can't do switch cases with handles.

Reply:
Thought you could coerce it to int primitive, but I guess not.